### PR TITLE
[Contribution guide] Add details on contributing

### PR DESCRIPTION
**Pull request by tcNickolas**
* Add sections on kata structure and on workbooks
* Add example pull requests for different types of contributions

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -7,10 +7,12 @@ We're so glad you asked!
...
 * [Improving Documentation](#improving-documentation)
 * [Contributing Code](#contributing-code)
-   * [Improving existing katas](#improving-existing-katas)
+   * [The structure of a kata/tutorial](#the-structure-of-a-katatutorial)
+   * [Adding new tasks to existing katas](#adding-new-tasks-to-existing-katas)
+   * [Contributing workbooks](#contributing-workbooks)
    * [Contributing new katas](#contributing-new-katas)
    * [Style guide](#style-guide)
-   * [Updating the Katas to the new QDK version](#updating-the-Katas-to-the-new-QDK-version)
+   * [Updating the Katas to the new QDK version](#updating-the-katas-to-the-new-qdk-version)
```
Comment (msoeken) [inline .github/CONTRIBUTING.md:15]:
```suggestion
   * [Updating the katas to the new QDK version](#updating-the-katas-to-the-new-qdk-version)
```

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -39,17 +41,40 @@ Whether you want to contribute a new task to an existing kata, to improve the te
...
 
-#### Contributing New Katas
+### Contributing workbooks
+
+A lot of katas and tutorials have *workbooks* - worked out solutions to the exercises presented in Jupyter Notebook format and cross-linked with the Jupyter Notebook "frontend" of the kata. 
+
+The explanations focus on the logical steps required to solve a problem; they illustrate the concepts that need to be applied to come up with a solution to the problem, explaining the mathematical steps required, and occasionally the Q# concepts and features used to implement the mathematical solution in the code. 
+The workbook should not be the primary source of knowledge on the subject matter; it assumes that the learner has already read a tutorial or a textbook and that is now seeking to improve your problem-solving skills. The learner should attempt solving the tasks of the respective kata first, and turn to the workbook only if stuck. While a textbook emphasizes knowledge acquisition, a workbook emphasizes skill acquisition.
```
Comment (msoeken) [inline .github/CONTRIBUTING.md:71]: In the first sentence _your_ does not match with _the learner_.

**Review by msoeken — APPROVED** [review]
Looks good to me.

File: .github/CONTRIBUTING.md (hunk 2 above)
Comment (tcNickolas) [reply .github/CONTRIBUTING.md:71]: Good catch, thank you - copied that one over from a workbook intro and missed the change in one place

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (tcNickolas) [reply .github/CONTRIBUTING.md:15]: This one is a bit subtle - "the Katas" is the project name, while "the katas" is plural of "a kata", a single set of exercises. If we say lowercase katas here, we'd have to say "and tutorials", and the version update we're doing is really across everything, including configuration files.

**Pull request merged**